django/contrib/admin/static/admin/js/selectfilter.js

```diff
@@ -0,0 +1,345 @@
+(function($) {
+    var settings = {
+        'name': '',
+        'verboseName': '',
+        'stacked': 0,
+        refresh_icons: function(selector) {
+            var available = selector.data('available');
+            var chosen = selector.data('chosen');
+            var is_from_selected = available.children().filter(':selected').length > 0;
+            var is_to_selected = chosen.children().filter(':selected').length > 0;
+            // Active if at least one item is selected
+            selector.find('.selector-add').toggleClass('active', is_from_selected);
+            selector.find('.selector-remove').toggleClass('active', is_to_selected);
+            // Active if the corresponding box isn't empty
+            selector.find('.selector-chooseall').toggleClass('active', available.find('option').length > 0);
+            selector.find('.selector-clearall').toggleClass('active', chosen.find('option').length > 0);
+        },
+        refresh_state: function(selector) {
+            settings.refresh_icons(selector);
+            var available = selector.data('available');
+            var chosen = selector.data('chosen');
+
+            // Re-generate the content of the actual, hidden, box.
```

Comment:
Regenerate
"actual, hidden, box" is a bit confusing -- maybe a high level overview of how the code works would be helpful